### PR TITLE
Add code formatter without rules

### PR DESCRIPTION
**Pull request by reacheight**

### CODE REVIEW

File: ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    {
+      FormatInsertedNodes(first, last, false);
+
+      FormatterImplHelper.CheckForMinimumSeparator(this, first, last);
+    }
+
+    public override void FormatDeletedNodes(ITreeNode parent, ITreeNode prevNode, ITreeNode nextNode)
+      => FormatterImplHelper.FormatDeletedNodesHelper(this, parent, prevNode, nextNode, false);
+
+    public override string OverridenSettingPrefix => "// @formatter:";
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs]: Does this string actually work?

File: ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  public class FSharpCodeFormatter : CodeFormatterBase<FSharpFormatSettingsKey>
+  {
+    private readonly FSharpFormatterInfoProvider formatterInfoProvider;
+
+    private readonly ConcurrentDictionary<FormatterImplHelper.TokenTypePair, bool> glueingCache =
+      new ConcurrentDictionary<FormatterImplHelper.TokenTypePair, bool>();
+
+    public FSharpCodeFormatter(
+      FSharpLanguage language,
+      CodeFormatterRequirements requirements,
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs]: Nit-picking: please don't put each arg on a separate line, such style is not used anywhere in the solution.

**Comment by auduchinok** [comment]
@osdm Could you please look at the formatter code?

The code is extracted from #81 and we'd like to merge it before continuing the work on the actual formatter rules.

File: ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs (hunk 1 above)
Comment (reacheight) [reply ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs]: Yes, if setting has `EditorConfigEntryAlias` attribute, it's value can be overridden using this string.

**Comment by auduchinok** [comment]
@reacheight Thanks!

File: ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs (hunk 1 above)
Comment (osdm) [reply ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs]: EditorConfigEntryAlias is not necessary really. Every setting in settings key marked with EditorConfigKey attribute could be here

**Pull request merged**